### PR TITLE
chore(Matomo): update app version to v4.15.1, reduce warnings

### PR DESCRIPTION
**Pull request by pmig**
This PR updates Matomo to v4.15.1 and fixes various warnings as tracked by #220 

We should do some end to end testing with existing installations

### CODE REVIEW

File: operator/src/main/kotlin/eu/glasskube/operator/apps/matomo/dependent/MatomoDeployment.kt [hunk 1]
```diff
@@ -158,20 +146,42 @@ class MatomoDeployment : CRUDKubernetesDependentResource<Deployment, Matomo>(Dep
...
+                                // If the container is still unresponsive after 10 minutes, it will be restarted
+                                failureThreshold = 60
+                                httpGet {
+                                    path = PROBE_PATH
+                                    port = "http".intOrString()
                                 }
-                                volumeMount {
-                                    name = cronVolumeName
-                                    mountPath = cronDir
+                            }
```
Comment (kosmoz) [inline operator/src/main/kotlin/eu/glasskube/operator/apps/matomo/dependent/MatomoDeployment.kt:159]: Suggestion: Matomo should be fast enough that it doesn't need a startup probe. just a liveness probe is probably fine.

**Review by kosmoz — CHANGES_REQUESTED** [review]
Looks like a reasonable change.

I will do a quick test now.

File: operator/src/main/kotlin/eu/glasskube/operator/apps/matomo/MatomoReconciler.kt [hunk 2]
```diff
@@ -57,6 +65,12 @@ class MatomoReconciler(private val kubernetesClient: KubernetesClient) :
         "CRDs provided by the MariaDB Operator must be present on the cluster."
 
     override fun reconcile(resource: Matomo, context: Context<Matomo>): UpdateControl<Matomo> {
+        context.getSecondaryResources(PersistentVolumeClaim::class.java)
+            .filter { it.metadata.name.endsWith("-misc") }
+            .forEach {
+                kubernetesClient.persistentVolumeClaims().resource(it).delete()
```
Comment (kosmoz) [inline operator/src/main/kotlin/eu/glasskube/operator/apps/matomo/MatomoReconciler.kt:71]: Suggestion: I think a logging statement would be appropriate here.

**Comment by kosmoz** [comment]
My test worked without issues :tada:

File: operator/src/main/kotlin/eu/glasskube/operator/apps/matomo/dependent/MatomoDeployment.kt (hunk 1 above)
Comment (pmig) [reply operator/src/main/kotlin/eu/glasskube/operator/apps/matomo/dependent/MatomoDeployment.kt:159]: you are right - migrations will be performed in the init container so the startup probe doesn't make sense.

**Pull request merged**